backend/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -89,14 +90,21 @@ def _init_root(self, root):
         :return: None
         """
         try:
-            for pth in [root, os.path.join(root, "etc/zypp/repos.d")]:
+            for pth in [root, os.path.join(CACHE_DIR, "root"), os.path.join(root, "etc/zypp/repos.d")]:
                 if not os.path.exists(pth):
-                    os.makedirs(pth)
-        except PermissionError as exc:
+                    if pth == os.path.join(CACHE_DIR, "root"):
+                        # If the Zypper root is being created for the first time
+                        # we copy the system RPM database to get the already imported
+                        # repository GPG keys (SUSE keys)
+                        copytree(RPM_DATABASE, os.path.join(pth, "var/lib/rpm/"))
```

Comment:
Hmm, the trusted gpg keys in the installed system is different from the keys we trust for SUSE Manager.
And if the customer trust additional keys for mirroring, you do not want to trust them also on your SUSE Manager Server.

Not sure what you plan in future iterations, but this could be the workflow:

- init an empty RPM DB in root
- import the GPG keys from the provided keyring
- If you trust an gpg key, you should export it to the root keyring.

I am not sure if you have 1 root, or one for every channel. If the later is the case you need to re-initialize every channel.